### ISSUE TITLE
Integrate touchHLE code to run basic iOS apps

### ISSUE THREAD

**Issue by upintheairsheep** [darlinghq/darling#1416]
touchHLE is a project similar to Darling that excecutes an iPhone's CPU via dynarmic libraries, and uses a reimplementations of Apple frameworks written in Rust. It is specifically made for running old 32-bit apps (currently iPhone OS 2.x) and does not have any plans for 64-bit support. Darling could benefit by reusing some (or hopefully all) implementations via code conversion, and can use dynarmic to run 32-bit iOS apps, then try executing the iOS games supported by touchHLE and try to mantain parity with it.
https://github.com/hikari-no-yume/touchHLE/ (main source)
https://github.com/LennyKappa/touchHLE/tree/thread-changes-real (adds some new threads)
https://github.com/LennyKappa/touchHLE/tree/later-ios (attempts support for iPhone OS 3.x)
https://github.com/KiritoDv/touchHLE/tree/mania_trunk (adds a few new functions to support _Mystery Mania_)
https://github.com/ciciplusplus/touchHLE/branches/active (a myriad of branches that attempt support for various new games, usually the latest branch updated contains the improvements of all other branches)

https://github.com/ipasimulator/ipasim (another project similar to touchHLE but adds support for basic 64-bit iOS apps via WinObjC)

**Comment by facekapow** [comment]
Yes, we're aware of touchHLE (and, personally, I think it's super cool), but thank you. See #1361.
